js/ui/appDisplay.js

```diff
@@ -167,23 +174,184 @@ const BaseAppView = new Lang.Class({
 
         this._items = {};
         this._allItems = [];
+
+        this._repositionedView = null;
+        this.repositionedIconData = [ null, null ];
     },
 
     _keyFocusIn: function(actor) {
         // Nothing by default
     },
 
+    _createItemIcon: function(item) {
+        throw new Error('Not implemented');
+    },
+
+    _createItemForId: function(itemId) {
+        let appSystem = Shell.AppSystem.get_default();
+        let item = null;
+
+        if (IconGridLayout.layout.iconIsFolder(itemId))
+            item = Shell.DesktopDirInfo.new(itemId);
```

Comment:
As mentioned in my previous review, I think it would be cleaner to return directly here.